OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -236,20 +236,18 @@ public override bool HandleKeyPress(KeyInput e)
 				}
 				else if (key == Game.Settings.Keys.SelectUnitsByTypeKey)
 				{
-					var selectedTypes = World.Selection.Actors
+					var selectedClasses = World.Selection.Actors
 						.Where(x => x.Owner == player)
-						.Select(a => a.Info);
+						.Select(a => a.Trait<Selectable>().Class);
 
-					Func<Actor, bool> cond = a => a.Owner == player && selectedTypes.Contains(a.Info);
-					var tl = worldRenderer.Viewport.TopLeft;
-					var br = worldRenderer.Viewport.BottomRight;
-					var newSelection = SelectActorsInBox(World, tl, br, cond);
+					Func<Actor, bool> cond = a => a.Owner == player && selectedClasses.Contains(a.Trait<Selectable>().Class);
+					var newSelection = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight, cond);
 
-					if (newSelection.Count() > selectedTypes.Count())
+					if (newSelection.Count() > selectedClasses.Count())
 						Game.Debug("Selected across screen");
 					else
 					{
-						newSelection = World.ActorMap.ActorsInWorld().Where(cond);
+						newSelection = World.ActorMap.ActorsInWorld().Where(x => x.HasTrait<Selectable>() && x.Trait<Selectable>().Info.Selectable && cond(x));
```

Comment:
`cond` already does the relevant checks, so these additional ones aren't necessary.
It's worth mentioning that by using `.Trait<Selectable>()` everywhere you're walking a fine line, but I'm counting that the owner check will filter out any unselectable actors, and this will only crash when there is bogus YAML (an actor belongs to the player but doesn't have the `Selectable` trait).